### DISCUSSION TITLE
Predictions for UpliftTreeClassifier differ by version 0.14.1 vs. 0.15

### DISCUSSION THREAD

**Discussion by cheecharron** [uber/causalml#752] [Q&A]
I am attempting to create a single-tree model of a 4-treatment dataset with a continuous outcome using the UpliftTreeClassifier. In version 0.15, this attempt generates an interpretable tree but likely unusable predictions for each treatment group: Whereas my observed outcomes range from -20 to 20, the outputted predictions appear to be bound by 0 and 1--I assume these correspond to probabilities for a binary outcome. However, in version 0.14.1, the outputted predictions tend to follow the range of the observed values, but there is no variance in predictions across cases (i.e., no subgroups). This also seems like an invalid solution. Is there any way to produce a simple single-tree model with more than 2 treatment groups and a continuous outcome?

**Comment by jeongyoonlee** [comment]
1. `UpliftTreeClassifer` is a classifier for the binary outcome variable. If you use a non-binary outcome, it won't work as intended.
2. Its prediction outputs are the treatment effect estimates of the treatments over the control, and their ranges can differ from the range of the outcome variable.
3. Currently, `CausalTreeRegressor` doesn't support multiple treatments (#775). For a regressor with multiple treatments, can you use meta-learners instead?
